procrustes/test/test_psdp.py

```diff
@@ -218,3 +268,30 @@ def test_psdp_peng_non_full_rank():
     )
     assert_almost_equal(s, actual_result, decimal=2)
     assert_almost_equal(error, 0.0, decimal=2)
+
+
+def test_psdp_opt_non_full_rank():
+    r"""Test PSDP when the to be transformed matrix doesn't have full rank."""
+    a = np.array(
+        [
+            [0.3452, -0.9897, 0.8082, -0.1739, -1.4692, -0.2531, 1.0339],
+            [0.2472, -1.4457, -0.6672, -0.5790, 1.2516, -0.8184, -0.4790],
+            [-1.3567, -0.9348, 0.7573, 1.7002, -0.9627, -0.5655, 2.5222],
+            [1.6639, 0.6111, -0.1858, 0.0485, 0.1136, 0.1372, -0.0149],
+            [-0.1400, -0.3303, -0.2965, 0.0218, 0.0565, -0.1907, -0.2544],
+            [-1.2662, 0.1905, 0.3302, -0.4041, 1.1479, -1.4716, 0.0857],
+        ]
+    )
+    b = np.array(
+        [
+            [6.3043, -6.5364, 1.2659, -2.7625, -2.9861, -5.4362, 2.7422],
+            [-0.5694, -9.4371, -5.5455, -15.6041, 24.4958, -20.4567, -11.4576],
+            [-0.1030, 2.3164, 3.0813, 8.1280, -10.6447, 6.6903, 7.9874],
+            [8.1678, -4.5977, 0.0559, -2.6948, 1.1326, -6.5904, 2.0167],
+            [6.3043, -6.5364, 1.2659, -2.7625, -2.9861, -5.4362, 2.7422],
+            [-0.5694, -9.4371, -5.5455, -15.6041, 24.4958, -20.4567, -11.4576],
+        ]
+    )
+    res = psdp_opt(a=a, b=b)
+    error = res["error"]
+    assert_almost_equal(error, 0.0, decimal=2)
```

Comment:
Can we add the actual positive semi-definitive matrix?

Reply:
Same issue as above. @FanwangM